### ISSUE TITLE
Traffic

### ISSUE THREAD

**Issue by itrack** [valhalla/valhalla#4411]
Hi!

The traffic system seems poorly documented and hard to understand how it works. (at least it was hard for me 😅)
I'm not very familiar with Valhalla in detail, I'm more of a docker run kind of user.

I opened this topic to ask if you can explain me how I can integrate a traffic system into Valhalla. From what I understand so far, this is done using a csv, can you give me an example of a csv?

Can the integration be done during a certain period of the day/week, i.e. can the traffic be predictive?

Thank you very much!

**Comment by nilsnolde** [comment]
True, it's not very well documented in the repo. And documentation is hard, also this kind of documentation (on my side) has the lowest priority. Basically, IMO if a company can shell out the cash to acquire this kind of data, they can also pay for implementation/consultation from people who went through that big-ish effort themselves.

Some refs:
- https://github.com/valhalla/valhalla/pull/4259
- smth I wrote a while back: https://gis-ops.com/traffic-in-valhalla/
- there's also some repo showing it bit more which you should be able to find (can't remember the name)
- look at the tests, historical/predictive traffic is covered and shows much of its implementation

And the inevitable pitch: enquiry@gis-ops.com :smile:
